packages/fetchai/skills/generic_buyer/behaviours.py

```diff
@@ -161,6 +161,11 @@ def _start_processing(self) -> None:
     def teardown(self) -> None:
         """Teardown behaviour."""
 
+        self.processing = None
+        self.processing_time = 0.0
+        self.timedout.clear()
+        self.waiting.clear()
+
```

Comment:
please note: I change something other than a test here, this is the only place I did